### PR TITLE
z_ovl_bg_spot07_taki OK

### PR DESCRIPTION
**Pull request by mzxrules**
this was actually pretty easy to get

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c [hunk 1]
```diff
@@ -36,13 +37,88 @@ const ActorInit Bg_Spot07_Taki_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot07_Taki/BgSpot07Taki_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+void BgSpot07Taki_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot07Taki* this = THIS;
+    s32 pad;
+    CollisionHeader* colHeader = NULL;
+    DynaPolyActor_Init(&this->dyna, DPM_PLAYER);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c:49]:
```suggestion
    CollisionHeader* colHeader = NULL;

    DynaPolyActor_Init(&this->dyna, DPM_PLAYER);
```

File: src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c [hunk 2]
```diff
@@ -36,13 +37,88 @@ const ActorInit Bg_Spot07_Taki_InitVars = {
...
+            CollisionHeader_GetVirtual(&D_060038FC, &colHeader);
+        }
+        this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, thisx, colHeader);
+    }
+    this->actionFunc = BgSpot07Taki_DoNothing;
+}
+
+void BgSpot07Taki_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot07Taki* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c:64]:
```suggestion
    BgSpot07Taki* this = THIS;

    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
```

File: src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c [hunk 3]
```diff
@@ -36,13 +37,88 @@ const ActorInit Bg_Spot07_Taki_InitVars = {
...
+}
+
+void BgSpot07Taki_DoNothing(BgSpot07Taki* this, GlobalContext* globalCtx) {
+}
+
+void BgSpot07Taki_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot07Taki* this = THIS;
+    this->actionFunc(this, globalCtx);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c:72]:
```suggestion
    BgSpot07Taki* this = THIS;

    this->actionFunc(this, globalCtx);
```

File: src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c [hunk 4]
```diff
@@ -36,13 +37,88 @@ const ActorInit Bg_Spot07_Taki_InitVars = {
...
+}
+
+void BgSpot07Taki_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    u32 pad;
+    u32 frames;
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_spot07_taki.c", 169);
+    frames = globalCtx->gameplayFrames;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c:79]:
```suggestion
    u32 pad;
    u32 frames;

    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_spot07_taki.c", 169);
    frames = globalCtx->gameplayFrames;
```

File: src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c [hunk 5]
```diff
@@ -36,13 +37,88 @@ const ActorInit Bg_Spot07_Taki_InitVars = {
...
+    if (gSaveContext.linkAge == 0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c:80]:
```suggestion
    if (LINK_IS_ADULT) {
```

File: src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c [hunk 6]
```diff
@@ -36,13 +37,88 @@ const ActorInit Bg_Spot07_Taki_InitVars = {
...
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_spot07_taki.c", 191),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot07_Taki/BgSpot07Taki_Update.s")
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, ((frames * -1) & 0x7F), ((frames * 1) & 0x7F), 32, 32, 1,
+                                ((frames * 1) & 0x7F), ((frames * 1) & 0x7F), 32, 32));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot07_Taki/BgSpot07Taki_Draw.s")
+    if (gSaveContext.linkAge != 0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c:99]:
```suggestion
    if (LINK_IS_CHILD) {
```

**Review by AngheloAlf — CHANGES_REQUESTED** [review]
Nice one!
Just a few small things.

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c [hunk 7]
```diff
@@ -36,13 +37,92 @@ const ActorInit Bg_Spot07_Taki_InitVars = {
...
+
+void BgSpot07Taki_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot07Taki* this = THIS;
+    s32 pad;
+    CollisionHeader* colHeader = NULL;
+
+    DynaPolyActor_Init(&this->dyna, DPM_PLAYER);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    if (LINK_IS_ADULT) {
+        if (thisx->params == 0) {
```
Comment (louist103) [inline src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c:53]:
```suggestion
        if (this->dyna.actor.params == 0) {
```
If possible.

File: src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c [hunk 8]
```diff
@@ -36,13 +37,92 @@ const ActorInit Bg_Spot07_Taki_InitVars = {
...
+    u32 pad;
+    u32 frames;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_spot07_taki.c", 169);
+    frames = globalCtx->gameplayFrames;
+    if (LINK_IS_ADULT) {
+        func_80093D18(globalCtx->state.gfxCtx);
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_spot07_taki.c", 177),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        if (thisx->params == 0) {
```
Comment (louist103) [inline src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c:88]:
```suggestion
        if (this->dyna.actor.params == 0) {
```
Same here.

File: src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c [hunk 9]
```diff
@@ -36,13 +37,92 @@ const ActorInit Bg_Spot07_Taki_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot07_Taki/BgSpot07Taki_Update.s")
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, ((frames * -1) & 0x7F), ((frames * 1) & 0x7F), 32, 32, 1,
+                                ((frames * 1) & 0x7F), ((frames * 1) & 0x7F), 32, 32));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot07_Taki/BgSpot07Taki_Draw.s")
+    if (LINK_IS_CHILD) {
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, 255, 128);
+        if (thisx->params == 0) {
```
Comment (louist103) [inline src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c:105]:
```suggestion
        if (this->dyna.actor.params == 0) {
```

File: src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c [hunk 10]
```diff
@@ -36,13 +37,92 @@ const ActorInit Bg_Spot07_Taki_InitVars = {
...
+        } else {
+            gSPSegment(POLY_XLU_DISP++, 0x09,
+                       Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, frames * 0, ((frames * -1) & 0x7F), 32, 32, 1,
+                                        frames * 0, ((frames * -1) & 0x7F), 32, 32));
+            gSPSegment(POLY_XLU_DISP++, 0x0A,
+                       Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, frames * 0, ((frames * 3) & 0x1FF), 32, 128, 1,
+                                        frames * 0, ((frames * 3) & 0x1FF), 32, 128));
+            gSPDisplayList(POLY_XLU_DISP++, &D_06000BE0);
+        }
+    } else if (thisx->params == 0) {
```
Comment (louist103) [inline src/overlays/actors/ovl_Bg_Spot07_Taki/z_bg_spot07_taki.c:122]:
```suggestion
    } else if (this->dyna.actor.params == 0) {
```

**Review by louist103 — CHANGES_REQUESTED** [review]
Looks good.  Just one group of things.

**Review by fig02 — APPROVED** [review]

**Review by Roman971 — APPROVED** [review]

**Pull request merged**